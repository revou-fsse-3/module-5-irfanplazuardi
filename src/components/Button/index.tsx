interface Props {
  label: string;
  className?: string;
  onclick?: () => void;
}

const Button = ({ label, className, onclick }: Props) => {
  return (
    <button className={className} onClick={onclick}>
      {label}
    </button>
  );
};

export default Button;
