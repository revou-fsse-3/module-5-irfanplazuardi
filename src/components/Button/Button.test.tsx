import { render, fireEvent, waitFor } from "@testing-library/react";
import Button from ".";

describe("Button unit testing", () => {
  const mocking = {
    onSubmit: jest.fn(),
  };
  test("Render button correctly", () => {
    const document = render(<Button label={"Test Button"} />);
    const element = document.getByText("Test Button");
    expect(element).toMatchSnapshot();
  });

  test("Button shows the correct class name", () => {
    const document = render(<Button label={"Test Button"} className="Title" />);
    const element = document.getByText("Test Button");
    expect(element).toHaveClass("Title");
  });

  test("Button click works correctly", async () => {
    const document = render(<Button label={"Test Button"} onclick={mocking.onSubmit} />);
    const element = document.getByText("Test Button");
    fireEvent.click(element);

    await waitFor(() => {
      expect(mocking.onSubmit).toHaveBeenCalledTimes(1);
    });
  });
});
