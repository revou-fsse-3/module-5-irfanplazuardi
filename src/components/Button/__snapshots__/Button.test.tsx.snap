// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button unit testing Render button correctly 1`] = `
<button>
  Test Button
</button>
`;
